Documentation/AdvancedBuildInstructions.md

```diff
@@ -53,18 +59,71 @@ Many parts of the SerenityOS codebase have debug functionality, mostly consistin
 To toggle a build option, add it to the `cmake` command invocation with a `-D` prefix. To enable it, add `=ON` at the end, or add `=OFF` to disable it. The complete command should look similarly to this:
 
 ```console
-$ cmake ../.. -G Ninja -DPROCESS_DEBUG=ON -DENABLE_PCI_IDS_DOWNLOAD=OFF
+$ cmake -B Build/i686 -DPROCESS_DEBUG=ON -DENABLE_PCI_IDS_DOWNLOAD=OFF
 ```
 
-For the changes to take effect, SerenityOS needs to be recompiled and the disk image needs to be rebuilt.
+Note that changing CMake cache variables such as PROCESS_DEBUG and ENABLE_UNDEFINED_SANITIZER can be done on
+an existing build without reconfiguring completely, and their effect will take effect the next time SerenityOS is recompiled.
```

Comment:
It's not just news to me, Andreas was unaware of this CMake concept in his latest office hours :) (around 1:15:50 https://youtu.be/puTsBKAQzk4?t=4552).  pointed it out to him on discord and he said "aha, I'd wondered why CMake options appeared 'sticky'". 

So I definitely want to put some sort of note in that conveys that CMake build directories do indeed cache their options/variables and that you can reconfigure to change just the specified from the command line while keeping the rest intact.

Reply:
I'm in no way against documenting this, just how it is right now feels overly elaborate.

Pointing out the fact that there is `ccmake`/`cmake-gui` for browsing the cache would also be worth it.  
Maybe a separate section dedicated to this would be better than a blurb at the bottom of this section.

Reply:
Ah that''s a good point. I've personally never invoked `ccmake` or `cmake-gui` for browsing the cache though, so I don't think I'm the best person to describe how that works in a detailed section :thinking: . It's possible that some of the manual work that has gone into getting Meta/Lagom/Tools/ConfigureComponents to look pretty in `whiptail` could be replaced by just using `ccmake`.

Reply:
The TUI and GUI are pretty easy to use. You just need to invoke them on the binary directory:
```console
$ ccmake path/to/existing/binary/dir
```
or in the GUI's case, you have the usual GUI stuff to do that.

I don't know what whiptail is, but if you love ncurses TUIs then ccmake will be right up your alley, even if a bit basic.

Reply:
whiptail is a TUI dialog box creator, the way we're using it creates a KConfig-looking interface for trimming down the system components that get created.

https://user-images.githubusercontent.com/8388494/132270074-d3210628-b253-4518-ad61-cf4cbe70cd55.mp4

ConfigureComponents is basically just a way to set the CMake cache variables for `BUILD_COMPONENTNAME` within some organized groups like "REQUIRED" "OPTIONAL" and "RECOMMENDED".

anyway, I'll move the cache description to a separate section with a blurb about the builtin tools for manipulating it alongside using `cmake` directly. After that i'm gonna start rebasing and splitting commits up so we can get this guy out of draft status.